packages/html/src/fill-in.ts

```diff
@@ -60,17 +21,18 @@ function setValue(element: TextFieldElement, value: string): void {
  * @param element The element to fill in text in
  * @param value The text value to fill in
  */
-export function fillIn(element: TextFieldElement, value: string): void {
-  let originalValue = element.value;
+export async function fillIn(interactor: Interactor<TextFieldElement, any>, value: string): Promise<void> {
+  let originalValue
+  await interactor.perform((element) => originalValue = element.value);
 
-  element.focus();
+  await interactor.perform((element) => element.focus());
 
-  clearText(element);
-  enterText(element, value);
+  await clear(interactor);
+  await Keyboard.type(value);
 
   if(originalValue !== value) {
-    dispatchChange(element);
+    await interactor.perform(dispatchChange);
   }
 
-  element.blur();
+  await interactor.perform((element) => element.blur());
```

Comment:
The whole fillIn action must be synchronous unless the second interaction action after focus or clear will fail in some circumstances, like if you use `TextField({ value: 'Foo', focused: false }).fillIn('Bar')`